public/jira/derived/work-status/work-status.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module is repsonsible for determining the correct workType ("design", "dev", "qa", "uat")
+ * and statusType ("qa", "uat", "todo", "done", "blocked") for an issue.
+ */
+export const workType = ["design", "dev", "qa", "uat"] as const;
+export const workTypes = workType;
+
+export const workflowHappyPath = ["todo", "design", "dev", "qa", "uat", "done"];
+export const workflowUnhappyStatuses = ["blocked"];
+
+const inQAStatus = { qa: true, "in qa": true, "qa complete": true };
+const inPartnerReviewStatus = { "partner review": true, uat: true };
+const inDoneStatus = { done: true, cancelled: true };
+const blockedStatus = { blocked: true, delayed: true };
+
+export const inPartnerReviewStatuses = Object.keys(inPartnerReviewStatus);
+export const inIdeaStatus = { idea: true, "to do": true, open: true };
+export const inIdeaStatuses = Object.keys(inIdeaStatus);
+
+type WorkType = (typeof workType)[number];
+
+type Status =
+  | keyof typeof inQAStatus
+  | keyof typeof inPartnerReviewStatus
+  | keyof typeof inIdeaStatus
+  | keyof typeof inDoneStatus
+  | keyof typeof blockedStatus
+  | (string & {});
```

Comment:
I remember this pattern of adding an empty object in a union had a special functionality but I forgot it :D 
Do you mind explaining it please?

Reply:
Oh! Yeah this is so you get the union types for auto-complete but still have the flexibility of string

Reply:
Yes the auto-complete. Cool stuff, thanks